tests/unit/test_registry.py

```diff
@@ -561,3 +564,102 @@ def prepare_hooks():
         self.assertEqual([item for (item,) in sequence], [
             'wrapped',
         ])
+
+
+class ExtendedStepTest(unittest.TestCase):
+    """
+    Test extended step.
+    """
+
+    def setUp(self):
+        self.extended_step = ExtendedStep()
+        self.single_expression_placeholders = \
+            self.extended_step.single_expression_placeholders
+        self.multi_expression_placeholders = \
+            self.extended_step.multi_expression_placeholders
+
+        self.extended_step.register_placeholder('TEST1', 't1')
+        self.extended_step.register_placeholder('TEST2', 't2_1', 't2_2')
+
+    def test_register_placeholder(self):
+        """Test registering placeholders."""
+
+        self.single_expression_placeholders.update({'TEST': 't1'})
+        self.multi_expression_placeholders.update({'TEST2': ('t2_1', 't2_2')})
+
+        self.assertDictEqual(
+            self.extended_step.single_expression_placeholders,
+            self.single_expression_placeholders
+        )
+
+        self.assertDictEqual(
+            self.extended_step.multi_expression_placeholders,
+            self.multi_expression_placeholders
+        )
+
+    def test_replace_placeholders(self):
+        """Test replacing placeholders in string."""
```

Comment:
Do not test private methods. This does not show me how to use the custom placeholders, and does not guarantee that they will actually work (no one says `_replace_placeholders` is actually used).